src/Model/Item.php

```diff
@@ -1234,35 +1234,18 @@ private static function guid($item, $notify)
 
 	private static function contactId($item)
 	{
-		$contact_id = (int)$item["contact-id"];
-
-		if (!empty($contact_id)) {
-			return $contact_id;
-		}
-		Logger::log('Missing contact-id. Called by: '.System::callstack(), Logger::DEBUG);
-		/*
-		 * First we are looking for a suitable contact that matches with the author of the post
-		 * This is done only for comments
-		 */
-		if ($item['parent-uri'] != $item['uri']) {
+		if (!empty($item['uid']) && !Contact::isSharing($item['author-id'], $item['uid'])) {
+			return $item['author-id'];
+		} elseif (!empty($item['contact-id'])) {
```

Comment:
Why isn't the `contact-id` field authoritative? For performance reasons, I'd rather we didn't perform a query (in `Contact::isSharing`) each time we use this method on `uid != 0` items.

Reply:
I do not understand the question "Why isn't the contact-id field authoritative?", what do you mean with that?

Concerning performance. When we check once upon storing, we don't have to check every time when we are displaying it: https://github.com/friendica/friendica/blob/develop/include/conversation.php#L806

Reply:
Authoritative means that it should be the first source of knowledge. In this case, when you want to know the `contact-id`, I feel like this is the first field that should be checked for. If it's empty, then you can check for `uid` and `author-id` and the rest.

Reply:
Authoritative means that it should be the first source of knowledge. In this case, when you want to know the `contact-id`, I feel like this is the first field that should be checked for. If it's empty, then you can check for `uid` and `author-id` and the rest.

Reply:
At first we have to check if the author is sharing with us. If this isn't the case we shouldn't store the user's contact id. 

Contacts are in the user's contacts whenever we share with them or they are sharing with us. But when we are only sharing with them (but not reverse) then we should treat this contact like an unknown one. This is important for checks like the one that I linked to above.

So this check has to be first since it will override the value in that case. Only after this we will check if the contact id had been set before.

Reply:
What are the cases where the `contact-id` field would have been wrongly set and needs to be overriden a posteriori?

Reply:
In the protocol implementations we mostly only check if there is an entry for the contact url and the matching uid. And yes, we could implement this check at every place where we create an item - but making this centrally is much more reliable. I'm also thinking about setting that contact-id only in the item class.